src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -0,0 +1,265 @@
+            "TON", "TTO", "TUV", "TZA", "UGA", "VCT", "VGB", "VIR", "WSM", "ZAF", "ZMB", "ZWE" };
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(WRONG_WAY_INSTRUCTIONS,
+            MULTIDIRECTIONAL_INSTRUCTIONS);
+
+    private Set<String> leftDrivingCountries;
+
+    /**
+     * An enum of RoundaboutDirections
+     */
+    public enum RoundaboutDirection
+    {
+        CLOCKWISE,
+        COUNTERCLOCKWISE,
+        // Handles the case where multiple directions were found in the roundabout
+        MULTIDIRECTIONAL,
+        // Handles the case where we were unable to get any information about the roundabout's
+        // Direction or if the roundabout's geometry was malformed (concave).
+        UNKNOWN
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public MalformedRoundaboutCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.leftDrivingCountries = new HashSet<>(
+                Arrays.asList((String[]) configurationValue(configuration, "traffic.countries.left",
+                        LEFT_DRIVING_COUNTRIES_DEFAULT)));
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // We check that the object is an instance of Edge
+        return object instanceof Edge
+                // Make sure that the object has an iso_country_code
+                && object.getTag(ISOCountryTag.KEY).isPresent()
+                // Make sure that the edges are instances of roundabout
+                && JunctionTag.isRoundabout(object)
+                // Is not two-way
+                && !OneWayTag.isExplicitlyTwoWay(object)
+                // And that the Edge has not already been marked as flagged
+                && !this.isFlagged(object.getIdentifier())
+                // Make sure that we are only looking at master edges
+                && ((Edge) object).isMasterEdge();
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final String isoCountryCode = edge.tag(ISOCountryTag.KEY).toUpperCase();
+
+        // Get all edges in the roundabout
+        final List<Edge> roundaboutEdges = getAllRoundaboutEdges(edge);
+
+        // Get the direction of the roundabout
+        final RoundaboutDirection direction = findRoundaboutDirection(roundaboutEdges);
+
+        // Determine if the roundabout is in a left or right driving country
+        final boolean isLeftDriving = leftDrivingCountries.contains(isoCountryCode);
```

Comment:
This could be moved after the first `if` block below. It is not used between lines 115-119.